mgmtfn/k8splugin/driver_test.go

```diff
@@ -0,0 +1,189 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8splugin
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"runtime"
+	"testing"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+func TestNetSetup(t *testing.T) {
+	la := netlink.NewLinkAttrs()
+	la.Name = "testlinkfoo"
+	newName := "testlinknewname"
+	address := "192.168.68.68/24"
+	defGW := "192.168.68.1"
+	staticRoute := "192.168.32.0/24"
+
+	runtime.LockOSThread()
+	locked := true
+	defer func() {
+		if locked {
+			runtime.UnlockOSThread()
+		}
+	}()
+
+	globalNS, err := netns.Get()
+	if err != nil {
+		t.Fatalf("failed to get the global network namespace: %v", err)
+	}
+	defer func() {
+		netns.Set(globalNS)
+		globalNS.Close()
+	}()
+
+	newNS, err := netns.New()
+	if err != nil {
+		t.Fatal("failed to create new network namespace")
+	}
+	defer newNS.Close()
+
+	cmd := exec.Command("sleep", "9999")
```

Comment:
busybox sleep doesn't support it, but both centos and ubuntu support `sleep infinity` which is waht I'm guessing you really want here.